pkg/podmounter/mountoptions/mount_options.go

```diff
@@ -64,6 +63,40 @@ func Send(ctx context.Context, sockPath string, options Options) error {
 	return nil
 }
 
+// unixSocketDialRetryInterval is the interval between retries on retryable errors in [dialWithRetry].
+const unixSocketDialRetryInterval = 5 * time.Millisecond
+
+// dialWithRetry tries to connect to Unix socket `sockPath` with retries until `ctx` is cancelled or hits the deadline.
+// It retries on two errors:
+//   - [syscall.ENOENT] returned when the Unix socket does not exists,
+//     which might be the case until Mountpoint Pod calls [Recv].
+//   - [syscall.ECONNREFUSED] returned when the Unix socket exists, but no one accepts the connection yet,
+//     which might be the case between the calls [net.Listen] and [net.Listener.Accept] in [Recv].
+func dialWithRetry(ctx context.Context, sockPath string) (*net.UnixConn, error) {
+	var d net.Dialer
+	var unixConn *net.UnixConn
+
+	err := wait.PollUntilContextCancel(ctx, unixSocketDialRetryInterval, true, func(ctx context.Context) (bool, error) {
+		conn, err := d.DialContext(ctx, "unix", sockPath)
+		if err == nil {
+			// no error, stop retrying and return the Unix connection.
+			unixConn = conn.(*net.UnixConn)
+			return true, nil
+		}
+
+		if errors.Is(err, syscall.ENOENT) ||
+			errors.Is(err, syscall.ECONNREFUSED) {
```

Comment:
Can we change the indentation here? It's confusing that it's the same level as the actual body of the if

Reply:
It's `go fmt`'s format, I can either make it one line or keep like this.

Reply:
I feel one line would be better

Reply:
Updated with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/380/commits/e64586d7520cf7de986728255abbfcc3173ce462